geoh5py/objects/surveys/electromagnetics/base.py

```diff
@@ -162,10 +162,11 @@ def add_validate_component_data(self, name: str, data_block: list | dict):
             )
 
         if isinstance(data_block, list):
-            assert np.all([entry.parent == self for entry in data_block]), (
-                f"The list of values provided for the component '{name}' "
-                f"must contain {FloatData} belonging to the target survey."
-            )
+            if not np.all([entry.parent == self for entry in data_block]):
```

Comment:
you could create a unitest to augment coverage for the error ;-)

Reply:
done